asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
 
 \chapter{Preamble}
 
-The semantics of ASL are embodied in a function that takes as input an ASL program
-and returns all possible behaviours of that program.
+The semantics of ASL defines all valid behaviors of a given ASL specification.
+More precisely, an ASL specification is first parsed into an \emph{abstract syntax tree},
+or AST, for short. Second, a type checker analyzes the \emph{parsed AST} for correctness and, if successful,
+returns a \emph{static environment} and a \emph{typed AST}. Finally, tools such as interpreters and verifiers
+can operate over the typed AST, based on the definition of the semantics, to test and analyze a given specification.
+
+The abstract syntax, parsed AST, and typed AST are defined in the abstract syntax reference~\cite{ASLAbstractSyntaxReference}.
+The ASL type system, which defines the set of parsed ASTs that are considered valid and how to
+produce the corresponding typed ASTs and static environments is defined in the typing reference~\cite{ASLTypingReference}.
+
+\paragraph{Basic Semantic Concepts.}
+The ASL semantics is given by a set of \emph{semantic relations} between \emph{configurations}.
```

Comment:
I am not sure the name "configuration" is the best choice here - please choose another name, perhaps "state"?

Reply:
The term is taken from the book "Semantics with Applications" by Nielson&Nielson.
If you insist, I will change it to state.

Reply:
It's not about my insisting - I *believe* that ASL has some notion of configurations which has nothing to do with ours here, hence my suggestion this was the best name for this specific document. You could perhaps do the following:
1. change the name in our document to be "semantic configurations"
2. *and* explicitly say "We use the name "configuration" after Nielson&Nielson [cite "Semantics with Applications"]"

Reply:
I defined semantic configurations and cited Nielson&Nielson. By the way, the reason configuration, rather than state, is used is because a state is already used for what we call environment. The semantics framework K also uses the term configurations.